oauth/opaque_validator.py

```diff
@@ -58,31 +60,74 @@ def introspect_token(self, token):
                       data=params,
                       headers=headers)
 
+        response_content_type = req.headers.get("Content-Type", "text/plain").split(";")[0]
+        result = {}
+        cache_duration_header_value = req.headers.get("Cache-Duration", None)
+
+        if cache_duration_header_value:
+            # Turn 'public, max-age=31536000' into {'public': None, 'max-age': '31536000'}
+            cache_duration_parts = dict(
+                (part_values[0], None if len(part_values) == 1 else part_values[1])
+                for part_values in [part.strip().split("=") for part in cache_duration_header_value.split(",")])
+
+            if "public" in cache_duration_parts:
+                result["cache_timeout"] = int(cache_duration_parts["max-age"])
+
         if req.status_code == 200:
-            return json.loads(req.text)
+            if response_content_type == "application/json":
```

Comment:
continue to work with OAuth servers that don't support requests for `application/jwt`